Resolve #1136 -- LevelUp now uses StatsModifier

cabeca1143 opened the pull request:
-Changed direct stat changes to StatsModifiers.
-Fixed Total Mana being added to BaseValue Every level up.

Resolve #1136

cabeca1143 committed:
Update Stats.cs

-Changed direct stat changes to StatsModifiers.
-Fixed Total Mana being added Every level up.

danil179 commented on GameServerLib/GameObjects/Stats/Stats.cs:
```diff
@@ -214,23 +214,22 @@ public void LevelUp()
-            ManaRegeneration.BaseValue += ManaRegenerationPerLevel;
-
+            StatsModifier statsLevelUp = new StatsModifier();
+            statsLevelUp.HealthPoints.BaseBonus = HealthPerLevel;
```
Should it be `=` or did you mean `+=`? (as all the others are `+=`)
Alternatively, shouldn't all the others be `=` as you only declare the variable? (I assume the default value is 0?)

cabeca1143 replied:
Oh, it was supposed to be just =
Functionally = or += doesn't change anything in this case, but I'll be uploading the changes anyway.

danil179 reviewed (commented):
Some comments.

cabeca1143 committed:
Update Stats.cs

danil179 reviewed (approved):
Approved. I'm not sure if that resolves the issue as I didn't check. In case it doesn't a new issue will be submitted.

danil179 reviewed (changes requested):
Cannot merge. Tests fail:
```
LeagueSandbox.GameServerTests.Tests.TestStats.TestLevelUp() in C:\projects\gameserver\GameServerLibTests\Tests\TestStats.cs:line 62
```

cabeca1143 committed:
Update TestStats.cs

Changed the BaseValue to BaseBonus, since StatsModifiers doesn't change the BaseValue

danil179 reviewed (approved):
Approved. Didn't test. Should solve most of the problems. I suggest to check where in the code `BaseValue` is used and see if it should be affected by leveling up.

danil179 commented:
Refs #1144.
